### PR TITLE
feat(wallet): ReceiveModal dialog UI tweaks

### LINKED ISSUE #13386: Wallet: QR dialog design tweaks to layout and aesthetics
The existing designs for the QR code have been updated to make them aesthetically more pleasing, leverage more standard design system components as well as making network preference handling more extensible.

The main changes are as follows:
- General tweaks to alignment / balancing in padding
- White backing, replacing the old grey backing
- Account icon in middle of QR
- Alignment of the network tags to base of QR - this means they can wrap onto multiple lines (previous designs has 2 lists left and right of the WR)
- Smaller, cleaner copy button 
- Neater, narrower network list

Screenshot 2024-02-01 at 5 01 38 PM

Figma:
https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?type=design&node-id=20734-337595&mode=design&t=2O68lxNGG9g1b1tx-4

The previous design has been retained here for reference:
https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?type=design&node-id=21242-170489&mode=design&t=2O68lxNGG9g1b1tx-4

Screenshot 2024-02-01 at 5 01 01 PM

### PR DESCRIPTION
**Pull request by IvanBelyakoff**
Closes #13386

### Affected areas

Wallet->Click on address -> Share address with QR code dialog

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [x] update storybook app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/15627093/a0ba4fec-dd0e-42a1-abff-df92a9d9d3b2

https://github.com/status-im/status-desktop/assets/15627093/a5c0550e-737e-4293-8882-71f1333441b5

https://github.com/status-im/status-desktop/assets/15627093/f64d796e-c547-43fb-85fb-12ecc18a9084

NOTE: I added one more copy button as per design. I asked @benjthayer to revise the design of copy button because now we have at least 2 icon copy buttons variants. When Ben finishes, we can fix/remove one of the buttons if needed.

### CODE REVIEW

File: ui/app/AppLayouts/Wallet/popups/ReceiveModal.qml [hunk 1]
```diff
@@ -83,24 +94,39 @@ StatusModal {
...
+
+            RowLayout {
+                id: rowLayout
+
+                width: parent.width
+
+                StatusBaseText {
+                    Layout.leftMargin: Style.current.bigPadding
+                    Layout.preferredWidth: parent.width - copyButton.width
+                    Layout.fillWidth: true
```
Comment (saledjenic) [inline ui/app/AppLayouts/Wallet/popups/ReceiveModal.qml:110]: Why?

File: ui/app/AppLayouts/Wallet/popups/ReceiveModal.qml [hunk 2]
```diff
@@ -83,24 +94,39 @@ StatusModal {
...
-            icon.name: "copy"
-            type: StatusRoundButton.Type.Tertiary
-            onClicked: RootStore.copyToClipboard(d.visibleAddress)
+                CopyButtonWithCircle {
+                    id: copyButton
+
+                    Layout.rightMargin: Style.current.bigPadding
+                    Layout.preferredWidth: 32
+                    Layout.preferredHeight: 32
+                    Layout.fillWidth: true
```
Comment (saledjenic) [inline ui/app/AppLayouts/Wallet/popups/ReceiveModal.qml:125]: And here?

File: ui/app/AppLayouts/Wallet/popups/ReceiveModal.qml [hunk 3]
```diff
@@ -30,6 +32,13 @@ StatusModal {
     property bool switchingAccounsEnabled: true
     property bool changingPreferredChainsEnabled: true
 
+    property string qrImageSource: RootStore.getQrCode(d.visibleAddress)
+    property var getNetworkShortNames: function(chainIDsString) {
+        return RootStore.getNetworkShortNames(chainIDsString)
+    }
+
+    property var networksModel: RootStore
```
Comment (saledjenic) [inline ui/app/AppLayouts/Wallet/popups/ReceiveModal.qml:40]: If the name is `networksModel`, could we make it point to model, not entire store?

`property var networksModel: RootStore.layer1Networks`

**Review by saledjenic — APPROVED** [review]

File: ui/imports/shared/controls/CopyButtonWithCircle.qml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        restoreMode: Binding.RestoreBindingOrValue
+    }
+
+    Binding on color {
+        when: successCircleVisible && d.copied
+        value: Theme.palette.successColor2
+        restoreMode: Binding.RestoreBindingOrValue
+    }
+
+    Rectangle {
```
[the remark this answers is not in the document]
Comment (caybro) [reply ui/imports/shared/controls/CopyButtonWithCircle.qml:44]: Ah yes, I see! All good then

**Review by caybro — APPROVED** [review]
LGTM

**Pull request merged**